server/mongo_data_access.go

```diff
@@ -97,18 +152,54 @@ func (dal *mongoDataAccessLayer) Delete(id, resourceType string) error {
 	}
 
 	collection := dal.Database.C(models.PluralizeLowerResourceName(resourceType))
-	return convertMongoErr(collection.RemoveId(bsonID.Hex()))
+	err = collection.RemoveId(bsonID.Hex())
+
+	if err == nil && dal.hasInterceptorsForVerbAndType("DELETE", resourceType) {
+		// Although this is a delete operation we need to get the resource first so we can
+		// run any interceptors on the resource before it's deleted.
+		resource, err := dal.Get(id, resourceType)
+		if err == nil {
+			dal.invokeInterceptors("DELETE", resourceType, resource)
+		}
+	}
+	return convertMongoErr(err)
 }
 
 func (dal *mongoDataAccessLayer) ConditionalDelete(query search.Query) (count int, err error) {
+	resourceType := query.Resource
 	searcher := search.NewMongoSearcher(dal.Database)
-	queryObject := searcher.CreateQueryObject(query)
+	collection := dal.Database.C(models.PluralizeLowerResourceName(resourceType))
+
+	var queryObject bson.M
+
+	if dal.hasInterceptorsForVerbAndType("DELETE", resourceType) {
+		// get the resources that are about to be deleted
+		var bundle *models.Bundle
+		bundle, err = dal.Search(url.URL{}, query) // the baseURL argument here does not matter
 
-	collection := dal.Database.C(models.PluralizeLowerResourceName(query.Resource))
+		if err != nil {
+			return count, convertMongoErr(err)
+		}
```

Comment:
Ah...  OK -- good point that if the search fails, you can't do the conditional delete by assembling the IDs (like in lines 190-192).  My point was that the search is only needed to support the interceptor logic -- so if the search fails, you should fall back to the other conditional delete code.  This will require a slight restructuring of the function I guess.